pkgs/development/libraries/ffmpeg/generic.nix

```diff
@@ -653,9 +653,9 @@ stdenv.mkDerivation (finalAttrs: {
 
   # Set RUNPATH so that libnvcuvid and libcuda in /run/opengl-driver(-32)/lib can be found.
   # See the explanation in addOpenGLRunpath.
-  postFixup = optionalString stdenv.isLinux ''
-    addOpenGLRunpath $out/lib/libavcodec.so
-    addOpenGLRunpath $out/lib/libavutil.so
+  postFixup = optionalString (stdenv.isLinux && withLib) ''
+    addOpenGLRunpath ${placeholder "lib"}/lib/libavcodec.so
+    addOpenGLRunpath ${placeholder "lib"}/lib/libavutil.so
```

Comment:
@Atemu is there a reason we only patch these two libraries? Maybe we could just use `nativeBuildInputs = [ cudaPackages.autoAddOpenGLRunpath ]`

I haven't any preferences though

Reply:
Might be because only those two depend on anything GL? It was like this before I refactored it.

I'm not familiar with this hook; what does it do? From the name it seems cuda-specific, no?

Reply:
This hook basically adds `/run/opengl-driver[-32]/lib` to the binary's RPATH so it can access any opengl libraries such as `libcuda` and `libcuvid` can be found (and `libcuda` is needed for NVENC).

Also important to note that `addOpenGLRunpath` silently fails if it doesn't find the binary, if `autoAddOpenGLRunpath` can prevent that from happening again I think it'd be a better choice.